File: oerp_custom/overrides/item.py
import frappe

DEFAULT_REQUEST_TYPE = "Purchase"


def sync_item_group_from_service_type(doc, method=None):
	"""Item Group (labelled "Item Sub Category" on the form) mirrors the
	selected Service Type value directly, and Item Main Category
	(fieldname item_sub_category — named the opposite of its own label)
	is set to that Item Group's own parent in the Item Group tree.

	Note: item_group is a Link to Item Group, so the first part only
	succeeds if an Item Group named exactly the same as the Service Type
	already exists — otherwise the save fails with an invalid-link error.
	"""
	if not doc.custom_service_type:
		return

	doc.item_group = doc.custom_service_type
	doc.item_sub_category = frappe.db.get_value("Item Group", doc.item_group, "parent_item_group")


def sync_qty_level_to_reorder(doc, method=None):
    for row in doc.get("custom_item_qty_level") or []:
        if not row.store:
            continue

        existing = next(
            (r for r in (doc.get("reorder_levels") or []) if r.warehouse == row.store),
            None,
        )

        values = {
            "warehouse": row.store,
            "warehouse_group": row.store,
            "warehouse_reorder_level": row.minimum or 0,
            "warehouse_reorder_qty": row.maximum or 0,
            "material_request_type": DEFAULT_REQUEST_TYPE,
        }

        if existing:
            existing.update(values)
        else:
            doc.append("reorder_levels", values)

File: oerp_custom/public/js/item.js
// Item: Item Group (labelled "Item Sub Category" on the form) mirrors the
// selected Service Type value directly, and Item Main Category (fieldname
// item_sub_category — named the opposite of its own label) is set to that
// Item Group's own parent in the Item Group tree.
// oerp_custom.overrides.item.sync_item_group_from_service_type re-applies
// this on validate, so API/import-created Items follow the same rule.
//
// Note: item_group is a Link to Item Group, so this only sticks if an Item
// Group named exactly the same as the Service Type already exists.

frappe.ui.form.on("Item", {
	custom_service_type(frm) {
		if (!frm.doc.custom_service_type) {
			return;
		}

		frm.set_value("item_group", frm.doc.custom_service_type).then(() => {
			frappe.db
				.get_value("Item Group", frm.doc.item_group, "parent_item_group")
				.then((r) => {
					const parent_item_group = r.message && r.message.parent_item_group;
					if (parent_item_group) {
						frm.set_value("item_sub_category", parent_item_group);
					}
				});
		});
	},
});
